paddle/fluid/framework/ir/mkldnn/conv1x1_depthwise_conv_mkldnn_fuse_pass.cc

```diff
@@ -0,0 +1,297 @@
+    return false;
+  }
+
+  int dw_groups = dw_conv_op->HasAttr("groups")
+                      ? PADDLE_GET_CONST(int, dw_conv_op->GetAttr("groups"))
+                      : 1;
+  if (dw_groups != dw_conv_filter_dims[0] ||
+      dw_groups != dw_conv_filter_dims[1] || dw_groups != 1) {
+    VLOG(4) << "Depthwise groups should be eqaul to 1 and to ic and oc to "
+               "perform fuse.";
+    return false;
+  }
+
+  int conv_groups = dw_conv_op->HasAttr("groups")
+                        ? PADDLE_GET_CONST(int, dw_conv_op->GetAttr("groups"))
+                        : 1;
+  if (conv_groups != 1) {
+    VLOG(4) << "Conv groups should be eqaul to 1 to perform fuse.";
+    return false;
+  }
+
+  return true;
+}
+
+void Conv1x1DepthwiseConvOneDNNFusePass::FuseConvDepthWise(
+    const std::string& conv_type, bool with_bias, Graph* graph) const {
+  FusePassBase::Init(name_scope_, graph);
+  GraphPatternDetector gpd;
+  patterns::ConvDepthwiseConv conv_depthwise_conv_pattern(gpd.mutable_pattern(),
+                                                          name_scope_);
+  conv_depthwise_conv_pattern(conv_type, with_bias);
+  int found_conv_depthwise_conv_count = 0;
+
+  auto handler = [&](const GraphPatternDetector::subgraph_t& subgraph,
+                     Graph* g) {
+    VLOG(4) << "Fuse conv with depthwise conv";
+    GET_IR_NODE_FROM_SUBGRAPH(
+        conv_weights, conv_weights, conv_depthwise_conv_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(conv_out, conv_out, conv_depthwise_conv_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(depthwise_conv_weights,
+                              depthwise_conv_weights,
+                              conv_depthwise_conv_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(
+        depthwise_conv_out, depthwise_conv_out, conv_depthwise_conv_pattern);
+
+    GET_IR_NODE_FROM_SUBGRAPH(conv, conv, conv_depthwise_conv_pattern);
+    GET_IR_NODE_FROM_SUBGRAPH(
+        depthwise_conv, depthwise_conv, conv_depthwise_conv_pattern);
+
+    auto* conv_op = conv->Op();
+    auto* depthwise_conv_op = depthwise_conv->Op();
+
+    if (!conv_op->HasAttr("use_mkldnn") ||
+        !(PADDLE_GET_CONST(bool, conv_op->GetAttr("use_mkldnn"))) ||
+        (!depthwise_conv_op->HasAttr("use_mkldnn") ||
+         !(PADDLE_GET_CONST(bool, depthwise_conv_op->GetAttr("use_mkldnn"))))) {
+      VLOG(4) << "The conv1x1 + depthwise_conv fusion may happen only when "
+                 "oneDNN library is used.";
+      return;
+    }
+
+    if ((conv_op->HasAttr("mkldnn_data_type") &&
+         PADDLE_GET_CONST(std::string, conv_op->GetAttr("mkldnn_data_type")) !=
+             "float32") ||
+        (depthwise_conv_op->HasAttr("mkldnn_data_type") &&
+         PADDLE_GET_CONST(std::string,
+                          depthwise_conv_op->GetAttr("mkldnn_data_type")) !=
+             "float32")) {
+      VLOG(4) << "The conv1x1 + depthwise_conv fusion is implemented only on "
+                 "float32.";
+      return;
+    }
```

Comment:
isValidForFusing is more about checking operators' properties, mkldnn is checked before isValidForFusing because it is a onednn pass and this condition has to be met